tests/ert/unit_tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -469,11 +469,16 @@ async def test_ensure_multi_level_events_in_order(evaluator_to_use):
         # about realizations, the state of the ensemble up until that point
         # should be not final (i.e. not cancelled, stopped, failed).
         ensemble_state = snapshot_event.snapshot.get("status")
+        finished_event_received = False
         async for event in monitor.track():
+            if isinstance(event, EETerminated):
```

Comment:
Can we check if the ensemble_state has changed appropriately? Otherwise this looks like an improvement.